examples/STM32L475VG-DISCOVERY-IOT/BTLE_sensors_tof_demo/BTLE_sensors_tof_demo.ino

```diff
@@ -0,0 +1,427 @@
+  VL53L0X_RangingMeasurementData_t pRangingMeasurementData;
+
+  do
+  {
+    if(top_done == 0)
+    {
+      NewDataReady = 0;
+      status = sensor_vl53l0x->GetMeasurementDataReady(&NewDataReady);
+
+      if( status ){
+        Serial.println("GetMeasurementDataReady sensor failed");
+      }
+
+      if(NewDataReady)
+      {
+        status = sensor_vl53l0x->ClearInterruptMask(0);
+        if( status ){
+          Serial.println("ClearInterruptMask sensor failed");
+        }
+
+        status = sensor_vl53l0x->GetRangingMeasurementData(&pRangingMeasurementData);
+        if( status ){
+          Serial.println("GetRangingMeasurementData sensor failed");
+        }
+
+        if (pRangingMeasurementData.RangeStatus == 0) {
+          // we have a valid range.
+          distance_top = pRangingMeasurementData.RangeMilliMeter;
+        }else {
+          distance_top = 1200;
+        }
+
+        top_done = 1;
+      }
+    }
+  }while(top_done == 0);
+
+  // Launch gesture detection algorithm.
+  gesture_code = tof_gestures_detectTAP_1(distance_top, &gestureTapData);
+
+  // Check the result of the gesture detection algorithm.
+  switch(gesture_code)
+  {
+    case GESTURES_SINGLE_TAP:
+      Serial.println(" => Tap detected");
+      ret = true;
+      break;
+    default:
+      // Do nothing
+      break;
+  }
+  return ret;
+}
+
+/* get data from environmental sensor and send update on BTLE */
+void update_environment_data(){
+  float humidity, temperature;
+  float pressure, temperature_lps22hb;
+
+  HumTemp->GetHumidity(&humidity);
+  HumTemp->GetTemperature(&temperature);
+
+  PressTemp->GetPressure(&pressure);
+  PressTemp->GetTemperature(&temperature_lps22hb);
+
+  //Update environnemental data
+  SensorService.Temp_Update(temperature*10);
+  SensorService.Press_Update(pressure*100);
+  SensorService.Humidity_Update(humidity*10);
+}
+
+/**
+ * Update the motion data on BLE. This depends of user action in front of vl53l0x
+ * sensor.
+ * A tap will switch the axis to update and a swipe will rotate the cube
+ *
+ * @param  AxesRaw_t* p_axes
+ * @retval None
+ */
+void update_motion_data(AxesRaw_t* p_axes)
+{
+  // change axis to update on each tap detection
+  if(tap_detected()){
+    axis_to_update++;
+    if( axis_to_update > UPDATE_AXIS_Y ) axis_to_update = UPDATE_AXIS_X;
+    // test should be on Z axis. But for now, the application on smartphone
+    // doesn't do anything with Z axis...
+    switch(axis_to_update){
+      case UPDATE_AXIS_X:
+        Serial.println("A swipe will now move the cube on X axis");
+        break;
+      case UPDATE_AXIS_Y:
+        Serial.println("A swipe will now move the cube on Y axis");
+        break;
+      case UPDATE_AXIS_Z:
+        Serial.println("A swipe will now move the cube on Z axis");
+        break;
+    }
+  }
+
+  if(swipe_detected()){
+    count_swipe = 20;
+  }
+
+  if(count_swipe > 0){
+    switch(axis_to_update){
+      case UPDATE_AXIS_X:
+        p_axes->AXIS_X += 100;
```

Comment:
explain the 100 value choice

Reply:
Arbitrary value to move the cube. Same as "official" BTLE example

Reply:
again worth a comment .. I think if I ask questions, maybe other users will as well